torch/utils/data/sampler.py

```diff
@@ -92,6 +92,53 @@ def __iter__(self):
     def __len__(self):
         return self.num_samples
 
+class EnlargeLabelShufflingSampler(Sampler):
+    """
+        label shuffling technique aimed to deal with imbalanced class problem
+        without replacement, manipulated by indices.
+        All classes are enlarged to the same amount, so classes can be trained equally.
+        argument:
+        indices: indices of labels of the whole dataset
+        """
+    
+    def __init__(self, indices):
+        # mapping between label index and sorted label index
+        sorted_labels = sorted(enumerate(indices), key=lambda x: x[1])
+        uniq_labels = set(sorted_labels)
+        count = 1
+        count_of_each_label = []
+        tmp = -1
+        # get count of each label
+        for (x, y) in sorted_labels:
+            if y == tmp:
+                count += 1
+            else:
+                if tmp != -1:
+                    count_of_each_label.append(count)
+                    count = 1
+            tmp = y
+        count_of_each_label.append(count)
+        # get the largest count among all classes. used to enlarge every class to the same amount
+        largest = int(np.amax(count_of_each_label))
+        self.count_of_each_label = count_of_each_label
+        self.enlarged_index = []
+        
+        # preidx used for find the mapping beginning of arg "sorted_labels"
+        preidx = 0
+        for x in range(len(self.count_of_each_label)):
+            idxes = np.remainder(t.randperm(largest).numpy(), self.count_of_each_label[x]) + preidx
+            for y in idxes:
+                self.enlarged_index.append(sorted_labels[y][0])
+            
+            preidx += int(self.count_of_each_label[x])
+
+    def __iter__(self):
+        shuffle(self.enlarged_index)
+        return iter(self.enlarged_index)
+    
+    def __len__(self):
+        return np.amax(self.count_of_each_label)*len(self.count_of_each_label)
```

Comment:
Numpy is not a hard dependency of torch, and is not even imported in this file so I believe this will be an error

Reply:
Thank you for the suggestions! shuffle is from random package, t is torch. Do you think by only importing the nparray it will be good? (It's my first time trying to contribute something.. sorry for these problems)

Reply:
the alternative way you described also seem to be a good solution, but by doing that, maybe there is some data not trained? This implementation will iterate the whole dataset. Only some data might be trained more than once.